### PR TITLE
[Go] basics - replace about.md file with concept files

### PR DESCRIPTION
**Pull request by ErikSchierboom**
In [this issue](https://github.com/exercism/v3/issues/2293) we're describing an evolution of the specification of this repo.
These changes include, amongst others, replacing the contents of the `after.md` document with individual concept documents. There are two documents per concept:

1. An `about.md` file containing the description of the concept
1. A `links.json` file containing a set of useful links related to the concept. 

This looks as follows in the repo:

```
<track>
├── concepts
│   ├── <concept>
│   │   ├── about.md
│   │   └── links.json
│   └── ...
```

This PR applies this change to this exercise by: 
                    
1. Copying the existing `after.md` file's contents to the `about.md` file of each concept listed in the exercise's `concepts` section in the `config.json` file.
1. Extract the links from the `after.md` file and put them into the `links.json` file

Before merging this PR, please check that the contents of the concept documents makes sense.
This is especially true when the exercise unlocks multiple concepts, as then the concept documents should be updated to only contain the information relevant to that concept.

Note: to make reviewing easier, the PR has been split into three separate commits. Please squash this PR upon merging.

### CODE REVIEW

File: languages/go/concepts/variables/links.json [hunk 1]
```diff
@@ -1 +1,35 @@
...
+  {
+    "url": "https://golang.org/ref/spec#Return_statements",
+    "description": "return"
+  },
+  {
+    "url": "https://golang.org/ref/spec#Exported_identifiers",
+    "description": "public-vs-private"
+  },
+  { "url": "https://golang.org/ref/spec#Comments", "description": "comments" }
+]
```
Comment (oanaOM) [inline languages/go/concepts/variables/links.json:35]: @ErikSchierboom it looks like link to first-class-functinos was skipped and assignment is duplicated.

**Review by oanaOM — COMMENTED** [review]
@ErikSchierboom , It looks like the about and links files have been changed for several concepts exercises under this commit. Is this intended?

File: languages/go/concepts/variables/links.json (hunk 1 above)
Comment (ErikSchierboom) [reply languages/go/concepts/variables/links.json:35]: @oanaOM Totally possible. The link extraction script was just a coarse scraping script. It is totally possible that there are missing links.

**Comment by ErikSchierboom** [comment]
> @ErikSchierboom , It looks like the about and links files have been changed for several concepts exercises under this commit. Is this intended?

Yes. Quoting the PR's comment:

> These changes include, amongst others, replacing the contents of the after.md document with individual concept documents. There are two documents per concept:

As the `basics` exercise unlocks multiple concepts, there are multiple concept documents. It is the task of the Go maintainers to edit the concept documents to contain just the information of that concept. So I fully expect the Go maintainers to update the files in this PR :)

**Comment by ErikSchierboom** [comment]
As we move forward towards v3 launch, this PR now need merging as this PR makes your track adhere to the latest specification. Could you please check whether they’re ok and merge them? If you don’t have time, I’ll merge them mid-next week and we can tidy things up later.

**Review by tehsphinx — CHANGES_REQUESTED** [review]
This would need a bit of work as there are 3 concepts here and each of them should only get the part that is about that concept.

**Comment by ErikSchierboom** [comment]
@tehsphinx I've split things up a bit

File: languages/go/concepts/packages/about.md [hunk 2]
```diff
@@ -1 +1,29 @@
-TODO: add information on packages concept
+## Packages
```
Comment (tehsphinx) [inline languages/go/concepts/packages/about.md:1]: Want to keep the headline?

**Review by tehsphinx — APPROVED** [review]
LGTM: Thx @ErikSchierboom for taking the time!

File: languages/go/concepts/packages/about.md (hunk 2 above)
Comment (ErikSchierboom) [reply languages/go/concepts/packages/about.md]: No, that was a mistake. Good catch!

**Comment by ErikSchierboom** [comment]
Thanks for the reviews!

**Pull request merged**